### PR TITLE
LinalgRefactor - fix styles / add docs

### PR DESCRIPTION
**Pull request by OXPHOS**

### CODE REVIEW

**Comment by OXPHOS** [comment]
@karlnapf can you check this when you got time? thanks!

File: src/shogun/mathematics/linalg/GPUMemoryViennaCL.h [hunk 1]
```diff
@@ -54,6 +54,8 @@ struct GPUMemoryViennaCL : public GPUMemoryBase<T>
 
 	typedef viennacl::backend::mem_handle VCLMemoryArray;
 	typedef viennacl::vector_base<T, std::size_t, std::ptrdiff_t> VCLVectorBase;
+
+/** @see <a href="http://viennacl.sourceforge.net/doc/changelog.html"> */
```
Comment (karlnapf) [inline src/shogun/mathematics/linalg/GPUMemoryViennaCL.h]: yes!

File: src/shogun/mathematics/linalg/LinalgNamespace.h [hunk 2]
```diff
@@ -108,9 +108,7 @@ LinalgBackendBase* infer_backend(const Container<T>& a, const Container<T>& b)
 template <typename T, template <typename> class Container>
 Container<T> add(const Container<T>& a, const Container<T>& b, T alpha=1, T beta=1)
 {
-	REQUIRE(a.vlen == b.vlen, "Vectors should have same length! "
-			"a(%d) vs b(%d)\n", a.vlen, b.vlen);
-
+	REQUIRE(a.vlen == b.vlen, "Length of vector a(%d) doesn't match vector b(%d).\n", a.vlen, b.vlen);
```
Comment (karlnapf) [inline src/shogun/mathematics/linalg/LinalgNamespace.h]: minor: space after a and b

**Comment by karlnapf** [comment]
Good! Thanks!

**Pull request merged**